src/NetMPA.Carting/NetMPA.Carting.Dal/Repositories/CartRepository.cs

```diff
@@ -8,26 +8,33 @@
 
 namespace NetMPA.Carting.Dal.Repositories
 {
-    internal class CartRepository : ICartRepository
+    public class CartRepository : ICartRepository
     {
-        public Task<CartDao> AddAsync(CartDao cartDao)
+        private IDictionary<Guid, CartDao> cartDB;
+
+        public CartRepository(IDictionary<Guid, CartDao> cartDB)
+        {
+            this.cartDB = cartDB;
+        }
+
+        public Task Add(CartDao cartDao)
         {
-            throw new NotImplementedException();
+            return new Task(() => cartDB.Add(cartDao.Id, cartDao));
```

Comment:
A new task is created here but not started. Usually `Task.Run `methods is used for this.

Reply:
BTW unit test for DAL hungs because of this.